### PR TITLE
Gravatar support for user profile picture

### LINKED ISSUE #1474: Gravatar profile picture
## Is your feature request related to a problem?

We could make the app's UI a bit more "human" by putting a user's profile picture in the top right.

## Describe the solution you'd like

A pretty simple solution would be pulling in the [gravatar](https://en.gravatar.com) linked to the user's email address.

## Describe alternatives you've considered

We could add pictures uploaded by users to PostHog, but that'd be significantly more effort and doesn't seem worth it at the moment.

## Additional context

Noticed today that Docker Hub does this, but I've definitely seen lots of other services do it as well before.

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

This makes it so that the user's Gravatar is used in the top right corner and user dropdown if available. If not available, Gravatar returns a 404 and the usual placeholder with the first letter of the user's name is shown.
Resolves #1474.

Jane

### CODE REVIEW

**Comment by Twixes** [comment]
Now also shown in Members:

Members

**Comment by Twixes** [comment]
Why not in Invites too:
Invites

**Review by paolodamico — APPROVED** [review]
In general lgtm and seems to work as expected (a few minor nits). Though I would strongly suggest we cache these images in the backend and fetch them from there. This will make sure we can use the images in other places well (e.g. emails), and we also avoid exposing our users to a third-party service (particularly concerned with self-hosted). Thoughts?

File: package.json [hunk 1]
```diff
@@ -42,10 +42,12 @@
         "@posthog/plugin-scaffold": "0.2.12",
         "@posthog/simmerjs": "0.7.2-posthog.2",
         "@sentry/browser": "^6.0.4",
+        "@types/crypto-js": "^4.0.1",
         "antd": "^4.11.2",
         "babel-preset-nano-react-app": "^0.1.0",
         "chart.js": "^2.9.3",
         "core-js": "3.6.5",
+        "crypto-js": "^4.0.0",
```
[the remark this answers is not in the document]
Comment (Twixes) [reply package.json:50]: Yep, this is specifically imported in a way utilizing tree shaking to only load the single function we need :)

File: frontend/src/layout/navigation/TopNavigation.tsx [hunk 2]
```diff
@@ -30,11 +30,36 @@ import { LinkButton } from 'lib/components/LinkButton'
...
 import { UserType } from '~/types'
 import { CreateInviteModalWithButton } from 'scenes/organization/TeamMembers/CreateInviteModal'
+import MD5 from 'crypto-js/md5'
+
+export interface ProfilePictureProps {
+    userName?: string
+    userEmail?: string
+}
+
+export function ProfilePicture({ userName, userEmail }: ProfilePictureProps): JSX.Element {
```
[the remark this answers is not in the document]
Comment (Twixes) [reply frontend/src/layout/navigation/TopNavigation.tsx]: Agreed

**Comment by Twixes** [comment]
IMO the privacy downside in using Gravatar client-side is small enough, it's not a script or anything. And airgapped deployments are unaffected by this PR. I've filed #3595 though to solve this server-side in due time, possibly when we finally get the `/api/user` DRF refactor done (right now I fear we'd introduce too much tech debt by working around the provisional user endpoint).

**Comment by mariusandra** [comment]
Gravatar is nice, though I think the biggest next win will be from getting avatars of users who log in with Google. :)

**Pull request merged**